### PR TITLE
Drop Python 2 support

### PR DESCRIPTION
**Pull request by nicodv**
See https://github.com/trevorstephens/gplearn/issues/104

Did I miss anything?

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,8 +2,6 @@ language: python
 
 env:
   matrix:
-    - PYTHON_VERSION="2.7" COVERAGE="true" LATEST="true"
```
Comment (trevorstephens) [inline .travis.yml]: My coverage is gone! Aieee!

**Comment by trevorstephens** [comment]
Appveyor for windows testing should drop Py2 as well me thinks. You ditch 2.7 from kmodes?

**Comment by trevorstephens** [comment]
I think I'd get rid of 3.4 at the same time

**Comment by nicodv** [comment]
Python 3.4 is still fairly heavily used, I believe, so I would wait with that.

And yes, doing the same for kmodes inspired this PR. Let's forcibly drag our users into the 21st century!

I addressed your comments, I believe.

**Comment by trevorstephens** [comment]
> Let's forcibly drag our users into the 21st century!

Haha!

I think sklearn is ditching 3.4 as well. I'll wait for them to move before I do it too though. LGTM, merging.

**Pull request merged**